applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nodal_reaction_response_function.cpp

```diff
@@ -0,0 +1,347 @@
+    {
+        KRATOS_TRY;
+
+        if (rResponseGradient.size() != rResidualGradient.size1())
+            rResponseGradient.resize(rResidualGradient.size1(), false);
+        rResponseGradient.clear();
+
+        for(IndexType i = 0; i < mpNeighborElements.size(); ++i)
+        {
+            Kratos::Element& ng_elem_i = mpNeighborElements[i];
+
+            if( rAdjointElement.Id() == ng_elem_i.Id() )
+            {
+                Vector filter_vector;
+                ProcessInfo process_info = rProcessInfo;
+                this->ConstructFilterVector(ng_elem_i, filter_vector, process_info);
+
+                Matrix left_hand_side;
+                ng_elem_i.CalculateLeftHandSide(left_hand_side, process_info);
+
+                noalias(rResponseGradient) = prod(left_hand_side, filter_vector);
+            }
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNodalReactionResponseFunction::CalculateFirstDerivativesGradient(
+        const Element& rAdjointElement,
+        const Matrix& rResidualGradient,
+        Vector& rResponseGradient,
+        const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+        rResponseGradient = ZeroVector(rResidualGradient.size1());
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNodalReactionResponseFunction::CalculateFirstDerivativesGradient(
+        const Condition& rAdjointCondition,
+        const Matrix& rResidualGradient,
+        Vector& rResponseGradient,
+        const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+        rResponseGradient = ZeroVector(rResidualGradient.size1());
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNodalReactionResponseFunction::CalculateSecondDerivativesGradient(
+        const Element& rAdjointElement,
+        const Matrix& rResidualGradient,
+        Vector& rResponseGradient,
+        const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+        rResponseGradient = ZeroVector(rResidualGradient.size1());
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNodalReactionResponseFunction::CalculateSecondDerivativesGradient(
+        const Condition& rAdjointCondition,
+        const Matrix& rResidualGradient,
+        Vector& rResponseGradient,
+        const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+        rResponseGradient = ZeroVector(rResidualGradient.size1());
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNodalReactionResponseFunction::CalculatePartialSensitivity(Element& rAdjointElement,
+                                             const Variable<double>& rVariable,
+                                             const Matrix& rSensitivityMatrix,
+                                             Vector& rSensitivityGradient,
+                                             const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+
+        rSensitivityGradient = ZeroVector(rSensitivityMatrix.size1());
+
+        if(!mPerformedModificationOfAdjointDisplacements)
```

Comment:
this is known also by checking the mAdjustInfluenceFunction. Its kind of a duplicated information

Reply:
I just want to ensure that the adjoint displacements were really modified. If  `mAdjustInfluenceFunction` is true but `FinalizeSolutionStep` was not executed than some sensitivities will be completely wrong. But I agree with you that having this second Boolean is not nice. I will think about erasing it.

Reply:
I see your point, but we are also not checking if e.g. the primals are calculated, we just assume they are there...

Reply:
Okay that is an argument. I will erase it.